sdk/common/src/main/java/io/opentelemetry/sdk/internal/AttributesMap.java

```diff
@@ -58,14 +58,22 @@ public static AttributesMap create(long capacity, int lengthLimit) {
    */
   @Override
   @Nullable
-  public Object put(AttributeKey<?> key, Object value) {
+  public Object put(AttributeKey<?> key, @Nullable Object value) {
+    if (value == null) {
+      return null;
```

Comment:
does this change have any effect on existing behaviors that we should be aware of?

Reply:
No it shouldn't. 

I wasn't quite remembering why I added this in the first place, so I retraced my steps and determined:

- `ExceptionAttributeResolver.AttributeSetter#setAttribute` accepts a `@Nullable T value`. We want this because we recently updated Span, LogRecord setAttribute methods to accept `@Nullable` values.
- AttributeMap#putIfCapacity acts as the implementation for this method
- AttributeMap#putIfCapacity delegates to AttributeMap#put (I want both methods to reduce accidental mistakes of bypassing the validation, as discussed in the javadoc of this class)